txpipe/maps.py

```diff
@@ -553,20 +558,15 @@ def run(self):
         # Convert count maps into density maps
         density_maps = []
         for i, ng in enumerate(ngal_maps):
-            mask_copy = mask.copy()
-            mask_copy[ng == healpy.UNSEEN] = 0
             ng[np.isnan(ng)] = 0.0
             ng[ng == healpy.UNSEEN] = 0
-            # Convert the number count maps to overdensity maps.
-            # First compute the overall mean object count per bin.
-            # mean clustering galaxies per pixel in this map
-            mu = np.average(ng, weights=mask_copy)
-            print(f"Mean number density in bin {i} = {mu}")
-            # and then use that to convert to overdensity
-            d = (ng - mu) / mu
-            # remove nans
-            d[mask == 0] = 0
-            density_maps.append(d)
+            delta_map = np.zeros(mask.shape, dtype=np.float64)
+            #delta_map[mask_copy>0] = ng[mask_copy>0] # Assuming they are already weighted by the mask
+            delta_map[mask>0] = ng[mask>0]/mask[mask>0] # Assuming that the weights do not include the mask
+            mu = np.mean(delta_map[mask>0])
+            delta_map[mask>0] = delta_map[mask>0]/mu-1
+            print(mu)
```

Comment:
you probably want to remove the print statement or add some description about what is printed (sorry, this is probably my fault).

Reply:
Removed!